test/components/link_test.exs

```diff
@@ -180,24 +180,79 @@ defmodule Surface.Components.LinkTest do
                ~s[<a data-method="put" data-to="/world" rel="nofollow" href="/world">hello</a>]
     end
 
+    test "link with :do contents" do
+      html =
+        render_surface do
+          ~H"""
+          <Link to="/hello">
+            {{ Phoenix.HTML.Tag.content_tag(:p, "world") }}
+          </Link>
+          """
+        end
+
+      assert html == """
+             <a href="/hello">
+               <p>world</p>
+             </a>
+             """
+    end
+
     test "link with scheme" do
-      html = render_surface(do: ~H[<Link to="/javascript:alert(<1>)" />])
-      assert html =~ ~s[<a href="/javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to="/javascript:alert(<1>)" />])
+      assert html =~ ~s[<a href="/javascript:alert(&lt;1&gt;)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:safe, "/javascript:alert(<1>)"} }} />])
-      assert html =~ ~s[<a href="/javascript:alert(<1>)"></a>]
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:safe, "/javascript:alert(<1>)"} }} />])
+
+      assert html =~ ~s[<a href="/javascript:alert(<1>)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, "alert(<1>)"} }} />])
-      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to={{ {:javascript, "alert(<1>)"} }} />])
+      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, 'alert(<1>)'} }} />])
-      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)"></a>]
+      html = render_surface(do: ~H[<Link label="foo" to={{ {:javascript, 'alert(<1>)'} }} />])
+      assert html =~ ~s[<a href="javascript:alert(&lt;1&gt;)">foo</a>]
+
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:javascript, {:safe, "alert(<1>)"}} }} />])
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, {:safe, "alert(<1>)"}} }} />])
-      assert html =~ ~s[<a href="javascript:alert(<1>)"></a>]
+      assert html =~ ~s[<a href="javascript:alert(<1>)">foo</a>]
 
-      html = render_surface(do: ~H[<Link to={{ {:javascript, {:safe, 'alert(<1>)'}} }} />])
-      assert html =~ ~s[<a href="javascript:alert(<1>)"></a>]
+      html =
+        render_surface(do: ~H[<Link label="foo" to={{ {:javascript, {:safe, 'alert(<1>)'}} }} />])
+
+      assert html =~ ~s[<a href="javascript:alert(<1>)">foo</a>]
+    end
+
+    test "link with invalid args" do
+      msg = "expected non-nil value for :to in <Link />"
```

Comment:
@msaraiva not too happy how this one turned out. The `:to` prop is marked as required and Surface already gives us a warning. Don't know how to make the Surface warning and the ArgumentError (raised in the Link update function) play nice together, any ideas?

Reply:
No need to add this one either. As you mentioned, `to` is already required so we got it covered.